Project/core/src/main/java/Model/Game.java

```diff
@@ -114,43 +119,64 @@ public Board getBoard() {
     }
 
     /**
-     * Gets the current number of recruited players revealed.
+     * Gets the total amount of recruits reveald so far.
      * 
-     * @return The amount of recruited players revealed.
+     * @return The total amount of recruits reveald so far.
      */
     public int getAmountRecruited() {
-        return this.amountRecruited;
+        int count = 0;
+        for (int[] recruits : this.recruitHistory) {
+            count += recruits[1];
+        }
+        return count;
     }
 
     /**
-     * Adds a specified amount to the number of recruited players.
+     * Gets the history of revealed recruits.
      * 
-     * @param newAmountRecruited The number of recruited players to add.
+     * @return A list of revealed recruites as (time, amount) pairs.
      */
-    public void addAmountRecruited(int newAmountRecruited) {
-        if (newAmountRecruited <= 0) {
+    public List<int[]> getRecruitHistory() {
+        return new ArrayList<>(this.recruitHistory); //Returns a copy for safety
+    }
+
+    /**
+     * Adds the number of recruits revealed during the current round.
+     * 
+     * @param amount The number of recruits revealed during current round.
+     */
+    public void addAmountRecruited(int amount) {
+        if (amount <= 0) {
             throw new IllegalArgumentException("Amount to add must be greater than 0.");
         }
-        this.amountRecruited += newAmountRecruited;
+        this.recruitHistory.add(new int[] { currentTime, amount });
     }
 
     /**
-     * Gets the current number of mindslips used in the game.
+     * Gets the mind slip history.
      * 
-     * @return The number of mindslips used.
+     * @return A list of times when mind slips were used.
      */
-    public int getMindSlipCount() {
-        return this.mindSlipCount;
+    public List<Integer> getMindSlipHistory() {
+        return new ArrayList<>(this.mindSlipHistory); //Returns a copy for safety
     }
 
     /**
-     * Increments the mind slip count by 1.
+     * Adds a mind slip event to the mind slip history.
+     * 
+     * @param time The time at which the mind slip was used.
      */
-    public void incrementMindSlipCount() {
-        if (this.mindSlipCount >= 2) {
-            throw new IllegalStateException("Mind slip count cannot exceed 2.");
+    public void addMindSlipEvent(int time) {
+        if (time < 1 || time >= maxTime) {
+            throw new IllegalArgumentException("Time is out of bounds.");
```

Comment:
Since the timeslip must be reveled directly once its used, would it not make kore sense to use the currentTime instead of passing a time parameter?